Current_SamplingFunctionality.cpp

```diff
@@ -0,0 +1,37 @@
+#include <iostream>
+#include <cstring>
+#include "Current_SamplingFunctionality.hpp"
+using namespace std;
+
+
+unsigned int ConsecutiveRanges(int Current_Samples[], int Current_Samples_size) {
+
+    unsigned int count = 1;
+    int idx; // index
```

Comment:
Avoid comments and name the variable itself as `index` instead of `idx`.
It saves a reference back to the declaration, everywhere we see the variable